data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/AbstractIndexManager.java

```diff
@@ -178,9 +180,16 @@ final boolean checkISMEnabled() throws IOException {
         final GetClusterSettingsRequest request = new GetClusterSettingsRequest.Builder()
                 .includeDefaults(true)
                 .build();
-        final GetClusterSettingsResponse response = openSearchClient.cluster().getSettings(request);
-        final String enabled = getISMEnabled(response);
-        return enabled != null && enabled.equals("true");
+        try {
+            final GetClusterSettingsResponse response = openSearchClient.cluster().getSettings(request);
+            final String enabled = getISMEnabled(response);
+            return enabled != null && enabled.equals("true");
+        } catch (OpenSearchException ex) {
+            if (NON_RETRYABLE_HTTP_STATUS.contains(ex.status())) {
```

Comment:
Should we have a stronger check here? This has any non-retryable indicating the ISM is disabled. What if the user lacks access? We might want that error to halt the sink.